packages/devextreme/testing/testcafe/helpers/testAccessibility.ts

```diff
@@ -0,0 +1,94 @@
+import createWidget, { WidgetName } from './createWidget';
+import { a11yCheck, A11yCheckOptions, ElementContext } from './accessibilityUtils';
+import { isMaterialBased } from './themeUtils';
+
+export interface Options {
+  [key: string]: any[];
+}
+
+export interface Configuration {
+  component: WidgetName;
+  options?: Options;
+  a11yCheckConfig?: A11yCheckOptions;
+  selector?: ElementContext;
+  created?: (t?: any, optionConfiguration?: Options) => Promise<void>;
+  after?: (t?: any, optionConfiguration?: Options) => Promise<void>;
+}
+
+const defaultSelector = '#container';
+const defaultOptions = {};
+const defaultCreated = async () => {};
+const defaultAfter = async () => {};
+const defaultA11yCheckConfig = isMaterialBased() ? {
+  runOnly: 'color-contrast',
+} : {};
+
+const generateConfigurations = (
+  options: Options,
+  index = 0,
+  prevConfigurations: Options[] = [],
+) => {
+  const keys = Object.keys(options);
+
+  if (index >= keys.length) {
+    return [Object.assign({}, ...prevConfigurations)];
+  }
+
+  const key = keys[index];
+  const values = options[key];
+
+  let configurations: object[] = [];
+
+  values.forEach((value) => {
+    const currentConfigurations = [
+      ...prevConfigurations,
```

Comment:
This destructuring wastes a linear time on each iteration. I suggest u getting rid of `prevConfigurations` and just move `push` before the recursive call and `pop` after it to improve the performance

Reply:
I think that is not applicable in the conditions of this decision.

Reply:
I don't see a reason why it's not applicable here. Did you try?
I mean smth like this:
```javascript
            const generateOptionMatrix = (
                options,
                index = 0,
                currentConfiguration = [],
            ) => {
                const keys = Object.keys(options);

                if (index === keys.length) {
                    return [Object.assign({}, ...currentConfiguration)];
                }

                const key = keys[index];
                const values = options[key];

                let configurationsMatrix = [];

                values.forEach((value) => {
                    currentConfiguration.push({ [key]: value });

                    const generatedConfigurations = generateOptionMatrix(
                        options,
                        index + 1,
                        currentConfiguration,
                    );
                    configurationsMatrix.push(...generatedConfigurations);

                    currentConfiguration.pop();                    
                });

                return configurationsMatrix;
            };
```